packages/react/src/ui/Connect/components/Network/NetworkDialog.tsx

```diff
@@ -39,7 +37,17 @@ export function NetworkDialog({
   const { isSupportedNetwork } = useIsSupportedNetwork();
   const { selectNetwork, isError, error, isPending } = useSelectNetwork();
   const { isConnected } = useIsConnected();
-  const [chainName, setChainName] = useState('');
+  const { data: chainName } = useQuery({
+    queryKey: ['chainName', networks[0]],
+    queryFn: async () => {
+      if (networks[0].url) {
+        const provider = await Provider.create(networks[0].url);
+        return provider.getChain().name;
+      }
+      return '';
+    },
+    initialData: '',
```

Comment:
otherwise apps that use `staleTime: Infinity` will never fetch this.

```suggestion
    placeholderData: '',
```